sdk/tdf_config.go

```diff
@@ -275,6 +281,81 @@ type TDFReaderConfig struct {
 
 	schemaValidationIntensity SchemaValidationIntensity
 	kasSessionKey             ocrypto.KeyPair
+	kasAllowlist              AllowList // KAS URLs that are allowed to be used for reading TDFs
+	ignoreAllowList           bool      // If true, the kasAllowlist will be ignored, and all KAS URLs will be allowed
+}
+
+type AllowList map[string]bool
+
+func getKasAddress(kasURL string) (string, error) {
+	// default to https if no scheme is provided
+	if !strings.Contains(kasURL, schemeSeperator) {
+		kasURL = schemeHTTPS + schemeSeperator + kasURL
+	}
+	parsedURL, err := url.Parse(kasURL)
+	if err != nil {
+		return "", fmt.Errorf("cannot parse kas url(%s): %w", kasURL, err)
+	}
+	if parsedURL.Hostname() == "" {
+		return "", fmt.Errorf("no host parsed from url: %s", kasURL)
+	}
+
+	// Default to "https" if no scheme is provided
+	if parsedURL.Scheme == "" {
+		parsedURL.Scheme = schemeHTTPS
+	}
+
+	// Default to port 443 if no port is provided
```

Comment:
I am unsure what the goal of this is, it seems like if the port is not set we should let the client choose the default based on the protocol (http vs https / 80 vs 443).  Why always default 443 if not set?

Reply:
the comment is incorrect, it only adds 443 if the scheme is https and there is no port provided. ill update the comment to reflect the logic